spec/requests/v1/items_request_spec.rb

```diff
@@ -155,16 +155,62 @@
 
     it "returns merchant data for a given item ID" do
         
-        get "/api/v1/items/#{@item1.id}/merchant"
+        get "/api/v1/items/#{@items[1]}/merchant"
     
         expect(response).to be_successful 
 
         merchant_items = JSON.parse(response.body, symbolize_names: true)
     
         expect(merchant_items[:data][:id]).to eq(@merchant.id.to_s)  
+    end 
+    
+    it "finds all items based on search criteria" do
+        get "/api/v1/items/find?name=#{@items[0].name}"
+
+        expect(response).to be_successful
+
+        found_items= JSON.parse(response.body, symbolize_names: true)
+
+        expect(found_items[:data][0][:id].to_i).to eq(@items[0].id)
+        expect(found_items[:data][0][:attributes][:name]).to eq(@items[0].name)
+
+
     end
 
-    
-    
+    it 'filter items by min_price' do
+        get '/api/v1/items/find?min_price=10.00'
+
+        expect(response).to be_successful
+
+        filter_items = JSON.parse(response.body, symbolize_names: true)
+
+        filter_items[:data].each do |item|
+            expect(item[:attributes][:unit_price]).to be >= 10.00
+        end
+    end
+
+    it 'filter items by max_price' do
+        get '/api/v1/items/find?max_price=10.00'
+
+        expect(response).to be_successful
+
+        filter_items = JSON.parse(response.body, symbolize_names: true)
+
+        filter_items[:data].each do |item|
+            expect(item[:attributes][:unit_price]).to be <= 20.00
```

Comment:
if the max price is $10 dollars than this assertion should check to make sure that all prices are under 10, not $20